src/clarity-angular/popover/signpost/signpost-positions.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Point } from "../common/popover";
+
+export interface Position {
+    anchorPoint: Point;
+    popoverPoint: Point;
+    offsetY: number;
+    offsetX: number;
+}
+
+
+export class SignpostPositions {
+    public default: Position = {
+        anchorPoint: Point.RIGHT_CENTER,
+        popoverPoint: Point.LEFT_CENTER,
+        offsetY: 6,
+        offsetX: 14
+    };
```

Comment:
So the problem here is that you're declaring a big constant, but you have to instantiate it every time so each popover creates a dozen different objects in memory. You want a shared constant.
On top of that, if you used the class names as keys, you could drastically reduce the bloating of the code in the component itself. So I recommend doing something like:

```typescript
export const SIGNPOST_POSITIONS: { [input: string]: Position } = {
    "top-left": {
        anchorPoint: Point.TOP_CENTER,
        popoverPoint: Point.BOTTOM_RIGHT,
        offsetY: -10,
        offsetX: 0
    },
    ...
}
```